compiler/src/main/scala/org/scalajs/core/compiler/PrepJSInterop.scala

```diff
@@ -1027,6 +1031,17 @@ abstract class PrepJSInterop extends plugins.PluginComponent
       ddef
     }
   }
+
+  private def checkInternalAnnotations(sym: Symbol): Unit = {
+    for {
+      ann <- compilerInternalAnnotations
```

Comment:
You'll have to make an exception for `@RawJSType` on `js.UndefOr` and `js.|`, I'm afraid. These two union types are implemented partly as a hack by artificially making them raw-js-type.

Alternative: have the compiler mark them `@RawJSType`.